spritzle/torrent.py

```diff
@@ -44,23 +46,28 @@ def __init__(self, core):
 
     def remove(self, torrent_handle, options=0):
         info_hash = str(torrent_handle.info_hash())
-        future_dict = self.delete_torrent_futures if options else self.remove_torrent_futures
-        if info_hash not in future_dict:
-            future_dict[info_hash] = asyncio.Future()
+
+        if info_hash not in self.remove_torrent_futures and info_hash not in self.delete_torrent_futures:
+            self.remove_torrent_futures[info_hash] = asyncio.Future()
+            if options & lt.options_t.delete_files:
+                self.delete_torrent_futures[info_hash] = asyncio.Future()
             self.core.session.remove_torrent(torrent_handle, options)
-        return future_dict[info_hash]
+
+        futures = []
```

Comment:
I think it might be a bit better to just create the futures list above, something like this:

```python
futures = []
if info_hash not in self.remove_torrent_futures and info_hash not in self.delete_torrent_futures:
    f = asyncio.Future()
    self.remove_torrent_futures[info_hash] = f
    futures.append(f)
    if options & lt.options_t.delete_files:
        f = asyncio.Future()
        self.delete_torrent_futures[info_hash] = f
        futures.append(f)
return asyncio.gather(*futures)
```

Reply:
The reason I did it this way is so that if remove was called twice before the alerts came back, the second call would still just wait on the first alerts. You think we don't need to worry about that case?

Reply:
Oh ok, that makes sense then.